### PR TITLE
feat(project): enhance project segment with .NET

### LINKED ISSUE #1568: Enhance dotnet segment to support .Net framework
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What would you like to see changed/added?

Enhance the dotnet segment to recognize if the folder contains a .Net framework project vs a dotnet core project.

Some suggestions are on the discussion such as parsing the project files and looking at the TargetFrameworkVersion element. 

I don't see an easy way to do this if only the .sln file is in the folder however, perhaps drilling into one of the projects in the solution and looking at that.

### PR DESCRIPTION
**Pull request by babieiev**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md]
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)

### Description

Enhance `project.go` by fetching current .NET project version.

Resolves #1568 

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/segments/project.go [hunk 1]
```diff
@@ -163,3 +176,29 @@ func (n *Project) getNuSpecPackage(item ProjectItem) (string, string) {
...
+		extension := filepath.Ext(file.Name())
+		if extension == ".csproj" || extension == ".fsproj" || extension == ".vbproj" {
+			name = strings.TrimSuffix(file.Name(), filepath.Ext(file.Name()))
+			content = n.env.FileContent(file.Name())
+			break
+		}
+	}
+	// the name of the parameter may differ depending on the version,
+	// so instead of xml.Unmarshal() we use regex:
+	tag := "(?P<tag><.*TargetFramework.*>(?P<tfm>.*)</.*TargetFramework.*>)"
```
Comment (JanDeDobbeleer) [inline src/segments/project.go]: Can we make `tag` and `tfm` uppercase? Let's not forget to also change below in that case.
```suggestion
	tag := "(?P<TAG><.*TargetFramework.*>(?P<TFM>.*)</.*TargetFramework.*>)"
```

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

**Comment by babieiev** [comment]
@JanDeDobbeleer it's strange that the `ubuntu` test fails because the lines are repeated three times. I didn't know that in `golang` everything is so strict with this 😅. Should I create local constants or do something else?

**Comment by JanDeDobbeleer** [comment]
@lesfiddler I'll check. One question I also have is that the behavior of this check still different compared to the others in project as they parse the project version number and not a target framework. Maybe we can extend the segment to have both values as both go and node definitely offer that. I bet there's also a `Version` property in the `Solution` file.

**Comment by babieiev** [comment]
> I bet there's also a `Version` property in the `Solution` file.

@JanDeDobbeleer unfortunately, I do not think that it is possible. Regarding the `.sln` file, @PilotBob already wrote in issue #1568:

> There is no distinction in a .sln file unfortunately.

Also, there is no information about this in the [official documentation](https://learn.microsoft.com/en-us/visualstudio/extensibility/internals/solution-dot-sln-file). Moreover, modern `dotnet` allows you to have projects without a `.sln` file. It is mostly needed to maintain the hierarchy of projects. The only version it contains is the Visual Studio version.

**Comment by babieiev** [comment]
> One question I also have is that the behavior of this check still different compared to the others in project as they parse the project version number and not a target framework.

I know that the behavior is not similar to any segment (that's why I suggested making a separate segment in [my previous PR](https://github.com/JanDeDobbeleer/oh-my-posh/pull/3005)). However, the project, in comparison with others, is the most suitable for this

**Comment by JanDeDobbeleer** [comment]
@lesfiddler alright, so I had a good thought on this and we'll need an additional Property called `.Target` here where we add this version number. I can extend the other languages like Node and go who have this as well.

**Comment by babieiev** [comment]
@JanDeDobbeleer if I understood it correctly, I should add a new property, but.. where exactly? `project.go` or `language.go`?

**Comment by JanDeDobbeleer** [comment]
@lesfiddler I'll do it. Don't worry, the PR is perfect, I can take over.

**Comment by JanDeDobbeleer** [comment]
@lesfiddler 👀

**Comment by babieiev** [comment]
@JanDeDobbeleer looks good. Do I need to do something with it?

**Comment by JanDeDobbeleer** [comment]
@lesfiddler nope, merging right away!

**Pull request merged**